File: javascript/socketd/src/socketd/transport/core/Asserts.ts
import type {Channel} from "./Channel";
import {Constants} from "./Constants";
import {SocketDChannelException, SocketDSizeLimitException} from "../../exception/SocketDException";

/**
 * 断言
 *
 * @author noear
 * @since 2.0
 */
export class Asserts {
    /**
     * 断言关闭
     */
    static assertClosed(channel: Channel | null) {
        if (channel != null && channel.closeCode() > 0) {
            throw new SocketDChannelException("This channel is closed, sessionId=" + channel.getSession().sessionId());
        }
    }

     static  isClosedAndEnd( channel:Channel) {
         return channel.closeCode() == Constants.CLOSE2009_USER
             || channel.closeCode() == Constants.CLOSE2008_OPEN_FAIL;
     }

    /**
     * 断言关闭
     */
    static assertClosedAndEnd(channel: Channel | null) {
        if (channel != null && Asserts.isClosedAndEnd(channel)) {
            throw new SocketDChannelException("This channel is closed, sessionId=" + channel.getSession().sessionId());
        }
    }

    /**
     * 断言 null
     */
    static assertNull(name: string, val: any) {
        if (val == null) {
            throw new Error("The argument cannot be null: " + name);
        }
    }

    /**
     * 断言 empty
     */
    static assertEmpty(name: string, val: string) {
        if (!val) {
            throw new Error("The argument cannot be empty: " + name);
        }
    }

    /**
     * 断言 size
     */
    static assertSize(name: string, size: number, limitSize: number) {
        if (size > limitSize) {
            const message = `This message ${name} size is out of limit ${limitSize} (${size})`;

            throw new SocketDSizeLimitException(message);
        }
    }
}

File: python/socketd/socketd/transport/core/Asserts.py
from socketd.exception.SocketDExecption import SocketDChannelException
from socketd.transport.core.Channel import Channel
from socketd.transport.core.Costants import Constants


class Asserts:
    @staticmethod
    def assert_size(name: str, size: int, limitSize: int) -> None:
        if size > limitSize:
            buf = f"This message {name} size is out of limit {limitSize} ({size})"
            raise RuntimeError(buf)

    @staticmethod
    def assert_closed(channel: Channel):
        if channel and channel.is_closed() > 0:
            raise SocketDChannelException("This channel is closed, sessionId=" + channel.get_session().session_id())

    @staticmethod
    def is_closed_and_end(channel: Channel):
        return channel.is_closed() == Constants.CLOSE2009_USER or channel.is_closed() == Constants.CLOSE2008_OPEN_FAIL

    @staticmethod
    def assert_closed_and_end(channel: Channel):
        if channel and Asserts.is_closed_and_end(channel):
            raise SocketDChannelException("This channel is closed, sessionId=" + channel.get_session().session_id())

    @staticmethod
    def assert_null(name: str, val):
        if val is None:
            raise SocketDChannelException("The argument cannot be null: " + name)

    @staticmethod
    def assert_empty(name: str, val: str):
        if val is None or len(val) == 0:
            raise SocketDChannelException("The argument cannot be empty: " + name)
